Document 'Brewfile.lock.json' in README (#1183)
Homebrew/homebrew-bundle: README.md

whalebrew "whalebrew/wget"
```

## Versions and lockfiles

Homebrew is a [rolling release](https://en.wikipedia.org/wiki/Rolling_release) package manager so does not support installing arbitrary older versions of software.
If your software needs specific pinned versions, consider [`whalebrew`](https://github.com/whalebrew/whalebrew) lines in your `Brewfile` to install [Docker](https://www.docker.com) containers).

After a successful `brew bundle` run, it creates a `Brewfile.lock.json` to record the environment. If a future `brew bundle` run fails, you can check the differences between `Brewfile.lock.json` to debug. As it can contain local environment information that varies between systems, it not worth committing to version control on multi-user repositories.

> Disable generation of the `Brewfile.lock.json` file by setting the environment variable with `export HOMEBREW_BUNDLE_NO_LOCK=1` or by using command-line argument `brew bundle --no-lock`.

Reviewer: ```suggestion
Disable generation of the `Brewfile.lock.json` file by setting the environment variable with `export HOMEBREW_BUNDLE_NO_LOCK=1` or by using the command-line argument `brew bundle --no-lock`.
```

whalebrew "whalebrew/wget"
```

> ## Lockfile

Reviewer 2: ```suggestion
## Versions and lockfiles
```

> Homebrew does not support installing specific versions of a library, only the most recent one, so there is no mechanism for installing specific versions given only a `.lock` file.
> 

Reviewer 2: ```suggestion
```

combining below

> If your software needs specific versions, consider [Vagrant](https://vagrantup.com/), Docker (perhaps with [whalebrew](https://github.com/whalebrew/whalebrew), which homebrew-bundle supports), or another tool to install versions explicitly.

Reviewer 2: ```suggestion
Homebrew is a [rolling release](https://en.wikipedia.org/wiki/Rolling_release) package manager so does not support installing arbitrary older versions of software.
If your software needs specific pinned versions, consider [`whalebrew`](https://github.com/whalebrew/whalebrew) lines in your `Brewfile` to install [Docker](https://www.docker.com) containers).
```

> After bundle installation, homebrew-bundle creates a `Brewfile.lock.json`. This file is for information purposes only as a record of the last successful `brew bundle`. Its presence does not change the result or behavior of `brew bundle`, but it can be useful as a record of a previous success when a subsequent `brew bundle` fails. It's probably not useful to commit to `git` or other version control, as it can contain local system info that varies between systems.

Reviewer 2: ```suggestion
After a successful `brew bundle` run, it creates a `Brewfile.lock.json` to record the environment. If a future `brew bundle` run fails, you can check the differences between `Brewfile.lock.json` to debug. As it can contain local environment information that varies between systems, it not worth committing to version control on multi-user repositories.
```

whalebrew "whalebrew/wget"
```

> ## On Lockfiles

Reviewer 3: ```suggestion
## Lockfile
```

Reviewer 2: ```suggestion
```

Let's cut this entirely

> Homebrew does not support installing specific versions of a library, only the most recent one, so there is no good mechanism for installing specific versions from a `.lock` file.

Reviewer 4: ```suggestion
Homebrew does not support installing specific versions of a library, only the most recent one, so there is no mechanism for installing specific versions given only a `.lock` file.
```

Reviewer 2: ```suggestion
```

Combining it below

> If your software needs specific versions then perhaps you'll want to look at using [Vagrant](https://vagrantup.com/) to better match your development and production environments.

Reviewer 4: While we're in here…
```suggestion
If your software needs specific versions, consider [Vagrant](https://vagrantup.com/), Docker (perhaps with [whalebrew](https://github.com/whalebrew/whalebrew), which homebrew-bundle supports), or another tool to install versions explicitly.
```

Reviewer 2: ```suggestion
Homebrew is a [rolling release](https://en.wikipedia.org/wiki/Rolling_release) package manager so does not support installing arbitrary older versions of software.
If your software needs specific pinned versions, consider [`whalebrew`](https://github.com/whalebrew/whalebrew) lines in your `Brewfile` to install [Docker](https://www.docker.com) containers).
```

Let's combine these a bit and remove mention of Vagrant which no-one seems to use any more.

> homebrew-bundle does create a `Brewfile.lock.json` for information purposes only, as a record of the last succesful `brew bundle`. It's presence does not change the result of `brew bundle`, but it can be useful as a record if a subsequent `brew bundle` fails. It's probably not useful to commit to git or other version control, as it can contain local system info that varies between systems.

Reviewer 4: ```suggestion
After bundle installation, homebrew-bundle creates a `Brewfile.lock.json`. This file is for information purposes only as a record of the last successful `brew bundle`. Its presence does not change the result or behavior of `brew bundle`, but it can be useful as a record of a previous success when a subsequent `brew bundle` fails. It's probably not useful to commit to `git` or other version control, as it can contain local system info that varies between systems.
```

Reviewer 2: ```suggestion
After a successful `brew bundle` run, it creates a `Brewfile.lock.json` to record the environment. If a future `brew bundle` run fails, you can check the differences between `Brewfile.lock.json` to debug. As it can contain local environment information that varies between systems, it not worth committing to version control on multi-user repositories.
```

> Generation of the `Brewfile.lock.json` file can be disabled with environment `HOMEBREW_BUNDLE_NO_LOCK=1` or by command-line argument `brew bundle --no-lock`.

Reviewer 4: ```suggestion
Disable generation of the `Brewfile.lock.json` file by setting environment variable `HOMEBREW_BUNDLE_NO_LOCK=1` or using command-line argument `brew bundle --no-lock`.
```

Reviewer 2: ```suggestion
Disable generation of the `Brewfile.lock.json` file by setting the environment variable with `export HOMEBREW_BUNDLE_NO_LOCK=1` or by using command-line argument `brew bundle --no-lock`.
```

Mild tweak
